src/test/java/uk/gov/justice/probation/courtcaseservice/listener/ProbationOffenderEventsListenerIntTest.java

```diff
@@ -0,0 +1,107 @@
+import com.amazonaws.services.sns.model.PublishResult;
+import com.amazonaws.services.sqs.model.PurgeQueueRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.testcontainers.shaded.com.fasterxml.jackson.core.JsonProcessingException;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import uk.gov.justice.probation.courtcaseservice.BaseIntTest;
+import uk.gov.justice.probation.courtcaseservice.jpa.entity.OffenderEntity;
+import uk.gov.justice.probation.courtcaseservice.jpa.entity.OffenderProbationStatus;
+import uk.gov.justice.probation.courtcaseservice.jpa.repository.OffenderRepository;
+import uk.gov.justice.probation.courtcaseservice.service.UserAgnosticOffenderService;
+import uk.gov.justice.probation.courtcaseservice.testUtil.OffenderEvent;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ProbationOffenderEventsListenerIntTest extends BaseIntTest {
+
+    @SpyBean
+    UserAgnosticOffenderService offenderService;
+
+    @SpyBean
+    OffenderRepository offenderRepository;
+
+    ObjectMapper objectMapper;
+    @Captor
+    ArgumentCaptor<String> crnArgumentCaptor;
+
+    @Captor
+    ArgumentCaptor<OffenderEntity> offenderEntityArgumentCaptor;
+
+
+    @BeforeEach
+    public void setUp() {
+        objectMapper = new ObjectMapper();
+        getOffenderEventReceiverQueueSqsClient().purgeQueue(new PurgeQueueRequest(getOffenderEventReceiverQueueUrl()));
+    }
+
+    @Test
+    public void shouldProcess_OffenderEventChangedMessage_AndUpdateOffenderProbationStatus() throws JsonProcessingException, InterruptedException {
+
+        var offenderEntity = OffenderEntity.builder()
+                .crn("X320741")
+                .probationStatus(OffenderProbationStatus.NOT_SENTENCED)
+                .awaitingPsr(false)
+                .breach(false)
+                .preSentenceActivity(false)
+                .build();
+
+        when(offenderRepository.findByCrn("X320741")).thenReturn(Optional.ofNullable(offenderEntity));
+
+        OffenderEvent sentencedEvent = OffenderEvent.builder()
+                .crn("X320741")
+                .offenderId(1L)
+                .nomsNumber("n123")
+                .sourceId(2L)
+                .eventDateTime(LocalDateTime.now())
+                .build();
+
+        var result = publishOffenderEvent(sentencedEvent);
+
+        assertThat(result.getSdkHttpMetadata().getHttpStatusCode()).isEqualTo(200);
+        assertThat(result.getMessageId()).isNotNull();
+
+        Thread.sleep(2000);
+       // assertOffenderEventReceiverQueueHasProcessedMessages();
```

Comment:
The test fails when using awaitality but passed using thread.  Could you please have a look? @jhevans